src/index.js

```diff
@@ -1,5 +1,52 @@
+import traverse from 'traverse';
 import arrayToObject from './lib/array-to-object';
 
+function computeTree(store, obj, mutationType, getterType, rootPath) {
+  // eslint-disable-next-line no-param-reassign
+  obj = JSON.parse(JSON.stringify(obj));
```

Comment:
Why is this necessary?

Reply:
It's just a `lazy` way of cloning data structures. We need it since we're changing the original data structure to accomodate getters/setters that trigger store mutations, while the original data structure is stored in vuex `state`. Another way of doing it is to make the traversal algorithm create new objects instead of mutating it, so we don't need to use cloning strategies.

Reply:
I used this strategy myself in the past to clone objects and in certain situations I'm fine with it because sometimes we have to take shortcuts in our projects. But because this is a library used by hundreds of people, I do not feel comfortable with this "hack".